### PR TITLE
Removing light client from Ethereum bridge design

### PR DESCRIPTION
**Pull request by batconjurer**
Refactored to specs to new updated design. This design does not implement an Ethereum light client but only requires validators to reach consensus on confirmed events emitted from Ethereum smart contracts

### CODE REVIEW

File: src/architecture/interoperability/ethereum-bridge.md [hunk 1]
```diff
@@ -105,58 +121,15 @@ struct TransferFromEthereum {
...
-    /// the original message exceeds the minimum number required to 
-    /// consider the message confirmed.
-    pub fn is_confirmed(&self) -> bool {
-        self.latest_descendant.1 - self.height >= self.min_confirmations
-    }
-}
-```
+The internal transaction that includes new events into `/eth_msgs` must 
+also update submit a tx containing a `TransferFromEthereum` instance based
+on any newly seen blocks. Thus `/eth_block/$block_hash/seen = true`
```
Comment (james-chf) [inline src/architecture/interoperability/ethereum-bridge.md:129]: `$block_hash -> $msg_hash`

File: src/architecture/interoperability/ethereum-bridge.md [hunk 2]
```diff
@@ -25,69 +25,85 @@ more cumbersome we will also add a limit on how fast wrapped ETH can be
...
+We want to store events from the smart contracts of our bridge onto Namada.
+We need to have consensus on these events, we will only include those that 
+have been seen  and validated by at least 2/3 of the staking validators in
+the blockchain storage.
+
+A valid message should correspond to the ICS20 fungible token packet. That 
+is to say it should be of the form:
+```rust
+struct EthEvent {
+    denom: String,
```
Comment (james-chf) [inline src/architecture/interoperability/ethereum-bridge.md:38]: `denom` - should it be in the ICS20 format (`{ics20Port}/{ics20Channel}/{denom}`) or could this just be an Ethereum address?

**Review by james-chf — APPROVED** [review]

File: src/architecture/interoperability/ethereum-bridge.md [hunk 3]
```diff
@@ -25,69 +25,85 @@ more cumbersome we will also add a limit on how fast wrapped ETH can be
...
+    sender: String,
+    receiver: String,
+}
+```
+Validators should ignore improperly formatted events.
+
+Each event should have a list of the validators that have seen
 this block and the current amount of stake associated with it. This
 will need to be appropriately adjusted across epoch boundaries. However,
 once a block has been seen by 2/3 of the staking validators, it is locked into a
```
[the remark this answers is not in the document]
Comment (batconjurer) [reply src/architecture/interoperability/ethereum-bridge.md:49]: Yes, I always find the language for this cumbersome and am prone to laziness 😅

File: src/architecture/interoperability/ethereum-bridge.md [hunk 4]
```diff
@@ -25,69 +25,85 @@ more cumbersome we will also add a limit on how fast wrapped ETH can be
...
+
+A valid message should correspond to the ICS20 fungible token packet. That 
+is to say it should be of the form:
+```rust
+struct EthEvent {
+    denom: String,
+    amount: u256,
+    sender: String,
+    receiver: String,
+}
```
[the remark this answers is not in the document]
Comment (batconjurer) [reply src/architecture/interoperability/ethereum-bridge.md:42]: I was actually thinking that the full node watchers would keep a queue on their side and check the confirmations there. This struct only appears when the watcher sends something to the ledger as a vote extension. It should assumed that it reached it's confirmations at this point, even if the users specified something other than the protocol minimum

[the remark this answers is not in the document]
Comment (batconjurer) [reply src/architecture/interoperability/ethereum-bridge.md:42]: The nonce is a good idea though.

File: src/architecture/interoperability/ethereum-bridge.md [hunk 5]
```diff
@@ -25,69 +25,85 @@ more cumbersome we will also add a limit on how fast wrapped ETH can be
...
+event has been applied on the Namada side.
+
+Changes to `/eth_msgs` are only ever made by internal transactions crafted by
 validators deterministically from the aggregate of vote extensions for the last Tendermint
-round. That is, changes to `/eth_block` are calculated and applied during the `FinalizeBlock` stage 
+round. That is, changes to `/eth_msgs` are calculated and applied during the `FinalizeBlock` stage
 of ABCI++ for block `n`, and then included (and validated) in a transaction in block `n+1`.
-It should not be possible for `/eth_block` storage to be modified by transactions submitted
+It should not be possible for `/eth_msgs` storage to be modified by transactions submitted
 from outside the ledger.
```
[the remark this answers is not in the document]
Comment (batconjurer) [reply src/architecture/interoperability/ethereum-bridge.md:96]: In the finalize block of call of block `n+1`, each validator makes this transaction and applies it. It is deterministically producible from the headers put in the proposal so there is no need to crowd the block with these txs.

File: src/architecture/interoperability/ethereum-bridge.md (hunk 2 above)
Comment (batconjurer) [reply src/architecture/interoperability/ethereum-bridge.md:38]: I think we have some leeway here. I'm ok with not sticking strictly to the ICS20 format but just using an ethereum address or something to denote wei, which doesn't have an address on ethereum as far as I know.

File: src/architecture/interoperability/ethereum-bridge.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (james-chf) [reply src/architecture/interoperability/ethereum-bridge.md:49]: I was guessing the former but when I was still learning how the staking stuff works I might've guessed the latter :P

File: src/architecture/interoperability/ethereum-bridge.md (hunk 2 above)
Comment (james-chf) [reply src/architecture/interoperability/ethereum-bridge.md:38]: Yeah I'd prefer not using the ICS20 format (for this first iteration of the Eth bridge at least). Thinking about integration with IBC, if someone wanted to send a wrapped Ethereum asset on Namada over IBC to some other chain, Namada would be treated as the "source" chain which I think is fine - the asset being passed around over IBC being the wrapped asset on Namada. But that will be further down the line anyway.

This `denom` field might have a custom format then to represent things other than ERC20 (including maybe ERC721 eventually in this same bridge).

**Pull request merged**